src/middlewared/middlewared/plugins/jail.py

```diff
@@ -473,3 +475,34 @@ def _import(self, job, jail):
         IOCImage().import_jail(jail)
 
         return True
+
+    @accepts()
+    async def get_plugin_versions(self):
+        """
+        Fetches a list of pkg's from the http://pkg.cdn.trueos.org/iocage/
+        repo and returns a dictionary for each plugin
+        """
+        r_pkgs = urllib.request.urlopen('http://pkg.cdn.trueos.org/iocage/All')
+        r_plugins = json.load(
+            urllib.request.urlopen(
+                'https://raw.githubusercontent.com/freenas/'
+                'iocage-ix-plugins/master/INDEX')
+        ).keys()
+        pkgs = {}
+
+        for i in r_pkgs.readlines():
+            i = i.decode().split('"')
+
+            if len(i) < 2:
+                continue
+
+            pkg_list = i[1].rsplit('-', 1)
+            if len(pkg_list) == 2 and pkg_list[0] in r_plugins:
```

Comment:
This means plugin name has to match freebsd pkg name? That seems suboptimal.

Shouldnt we use the plugins manifest to grab the right freebsd matching pkg?

Reply:
That's exactly what that routine is doing :)

Reply:
It is not, read again what I said. Plugin manifest, not packages list.

Reply:
That's what https://raw.githubusercontent.com/freenas/iocage-ix-plugins/master/INDEX is, the plugin manifest's keys are pkg-names, as some of them don't have a package. Which reminds me we need to handle those in this PR.